pkg/performanceprofile/controller/performanceprofile/hypershift/components/handler.go

```diff
@@ -0,0 +1,314 @@
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	mcov1 "github.com/openshift/api/machineconfiguration/v1"
+	performancev2 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/performanceprofile/v2"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/config"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/components"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/components/machineconfig"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/components/manifestset"
+	profileutil "github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/components/profile"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/hypershift"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/resources"
+)
+
+const (
+	hypershiftPerformanceProfileNameLabel = "hypershift.openshift.io/performanceProfileName"
+	hypershiftNodePoolLabel               = "hypershift.openshift.io/nodePool"
+	tunedConfigMapLabel                   = "hypershift.openshift.io/tuned-config"
+	tunedConfigMapConfigKey               = "tuning"
+	mcoConfigMapConfigKey                 = "config"
+	ntoGeneratedMachineConfigLabel        = "hypershift.openshift.io/nto-generated-machine-config"
+)
+
+var _ components.Handler = &handler{}
+
+type handler struct {
+	controlPlaneClient client.Client
+	dataPlaneClient    client.Client
+	scheme             *runtime.Scheme
+}
+
+func NewHandler(controlPlaneClient, dataPlaneClient client.Client, scheme *runtime.Scheme) components.Handler {
+	return &handler{
+		controlPlaneClient: controlPlaneClient,
+		dataPlaneClient:    dataPlaneClient,
+		scheme:             scheme,
+	}
+}
+
+func (h *handler) Delete(ctx context.Context, profileName string) error {
+	// ConfigMap is marked for deletion and waiting for finalizers to be empty
+	// so better to clean up and delete the objects.
+
+	// delete RtClass, as right now all the other components created from this PP
+	// are embedded into ConfigMaps, which has an OwnerReference with the PP configmap
+	// and will be deleted by k8s machinery triggering the deletion procedure of the embedded elements.
+	rtName := components.GetComponentName(profileName, components.ComponentNamePrefix)
+	rtClass, err := resources.GetRuntimeClass(ctx, h.dataPlaneClient, rtName)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			// rtClass not found so nothing to delete, so delete process has finished
+			return nil
+		}
+		return fmt.Errorf("unable to read RuntimeClass %q, error: %w. Unable to finalize deletion procedure properly", rtName, err)
+	}
+
+	err = h.dataPlaneClient.Delete(ctx, rtClass)
+	return err
+}
+
+func (h *handler) Exists(ctx context.Context, profileName string) bool {
```

Comment:
this seems to tell the caller if _any_ object associated with the PerfProfile exist. Is it OK or do we need to ensure _all_ of them exist (for consistency?)

Reply:
1. It is OK, if one/multiple/all objects exists, it ended up with the same result of re-queuing  the request.
2. This implementation is similar to non-Hypershift